sklearn/utils/optimize.py

```diff
@@ -15,7 +15,11 @@
 
 import numpy as np
 import warnings
-from scipy.optimize.linesearch import line_search_wolfe2, line_search_wolfe1
+
+try:
+    from scipy.optimize._linesearch import line_search_wolfe2, line_search_wolfe1
+except ImportError:  # SciPy < 1.8
+    from scipy.optimize.linesearch import line_search_wolfe2, line_search_wolfe1
```

Comment:
It looks like `line_search` is `line_search_wolfe2`,

https://github.com/scipy/scipy/blob/d7fa27f0c4a33b6e4f7bafec6605961428ca6523/scipy/optimize/_optimize.py#L34

If we want to use the public API, we can import `line_search` as `line_search_wolfe2`, but I think what we have now is clearer.

Reply:
Shouldn't this try except be done as a utils.fixes ?

Reply:
I agree. To be consistent, we should place this in `sklearn/utils/fixes.py`.

Reply:
Done